UPM no longer builds with latest MRAA due to GPIO enums refactoring

Since https://github.com/intel-iot-devkit/mraa/commit/f080a744973d6687129cf2fd5e63175d46a753be GPIO enums have different names, which causes UPM to fail during building with errors like

```
<cut>
| /edison_dev/yocto/ww18/build/tmp/work/core2-32-poky-linux/upm/0.3.1+gitAUTOINC+cff584d8b9-r0/git/src/hcsr04/hcsr04.cxx: In constructor 'upm::HCSR04::HCSR04(uint8_t, uint8_t, void (*)(void*))':
| /edison_dev/yocto/ww18/build/tmp/work/core2-32-poky-linux/upm/0.3.1+gitAUTOINC+cff584d8b9-r0/git/src/hcsr04/hcsr04.cxx:52:5: error: 'gpio_edge_t' was not declared in this scope
|      gpio_edge_t edge = MRAA_GPIO_EDGE_BOTH;
|      ^
| /edison_dev/yocto/ww18/build/tmp/work/core2-32-poky-linux/upm/0.3.1+gitAUTOINC+cff584d8b9-r0/git/src/hcsr04/hcsr04.cxx:52:17: error: expected ';' before 'edge'
|      gpio_edge_t edge = MRAA_GPIO_EDGE_BOTH;
|                  ^
| /edison_dev/yocto/ww18/build/tmp/work/core2-32-poky-linux/upm/0.3.1+gitAUTOINC+cff584d8b9-r0/git/src/hcsr04/hcsr04.cxx:53:34: error: 'edge' was not declared in this scope
|      mraa_gpio_isr (m_echoPinCtx, edge, fptr, NULL);
|                                   ^
| make[2]: *** [src/hcsr04/CMakeFiles/jsupm_hcsr04.dir/hcsr04.cxx.o] Error 1
<cut>
```

I think I could prepare the patch for UPM to fix this, but I wonder what's your view on that, i.e. do you assume only released mraa versions as valid for building UPM? CMake config has `>=` only and I don't see any other mentions of this elsewhere.

EDIT: I've worked this around in my Bitbake recipe for UPM and it turned out it's just one place where these enums are used, the diff is below, so the patch will be trivial, as soon as we clarify what's your approach to mraa versions dependency.

```diff
diff --git a/src/hcsr04/hcsr04.cxx b/src/hcsr04/hcsr04.cxx
index aecd8a5..79609f0 100644
--- a/src/hcsr04/hcsr04.cxx
+++ b/src/hcsr04/hcsr04.cxx
@@ -49,7 +49,7 @@ HCSR04::HCSR04 (uint8_t triggerPin, uint8_t echoPin, void (*fptr)(void *)) {
     }
 
     mraa_gpio_dir(m_echoPinCtx, MRAA_GPIO_IN);
-    gpio_edge_t edge = MRAA_GPIO_EDGE_BOTH;
+    mraa_gpio_edge_t edge = MRAA_GPIO_EDGE_BOTH;
     mraa_gpio_isr (m_echoPinCtx, edge, fptr, NULL);
 }
```

--- a/src/hcsr04/hcsr04.cxx
+++ b/src/hcsr04/hcsr04.cxx
@@ -49,8 +49,7 @@ HCSR04::HCSR04 (uint8_t triggerPin, uint8_t echoPin, void (*fptr)(void *)) {
     }
 
     mraa_gpio_dir(m_echoPinCtx, MRAA_GPIO_IN);
-    gpio_edge_t edge = MRAA_GPIO_EDGE_BOTH;
-    mraa_gpio_isr (m_echoPinCtx, edge, fptr, NULL);
+    mraa_gpio_isr(m_echoPinCtx, MRAA_GPIO_EDGE_BOTH, fptr, NULL);
 }
 
 HCSR04::~HCSR04 () {

